### PR TITLE
Add persistent volume claim to prebuild settings

### PR DESCRIPTION
**Pull request by sagor999**
## Description

This adds setting to prebuilds to enable support for persistent volume claims.
Made sure to add a warning to let users know to not enable this setting, but I am open for a better alternative\text or if there is way to gate this feature to gitpod only for now.

This is how it would look:
Screenshot from 2022-06-08 16-41-29

## Related Issue(s)

Related: https://github.com/gitpod-io/gitpod/issues/10260

## How to test

Launch preview env, add your project to it and navigate to settings.

## Release Notes

```release-note
none
```

### CODE REVIEW

**Comment by roboquat** [comment]
@sagor999: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by easyCZ** [comment]
Hey @sagor999, this is a great use-case for FeatureFlags we've recently introduced. We've not got too much in the way of write up for this so I'm happy to pair initially to bring you up to speed. I'll then use this as the basis for some write-up to make it a bit more self-serve. Let me know when you're free or drop something in my calendar.

You can also have a look at existing usage in 

https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/src/Menu.tsx#L164-L182

**Comment by sagor999** [comment]
@easyCZ PTAL

**Review by easyCZ — APPROVED** [review]
Usage of the feature flag looks good to me, and in general PR looks good. I'm no expert in React however. Adding hold in case you want to get a review from someone else as well.

/hold

**Comment by sagor999** [comment]
@jankeromnes would you be able to take a look at React portion?

**Comment by geropl** [comment]
@sagor999 Did you think about replacing that project-level config with a feature-flag entirely? :thinking: Or is it that you really want to expose this option to users? :thinking:

**Comment by sagor999** [comment]
@geropl yes, I would like it being exposed to users, as we will slowly start offering this to some of our SaaS customers to try it out, if we think it will be a great benefit to them and they are willing to test it out. 
I also want to be able to switch it on and off for a project without having to do it in configcat.

**Pull request merged**
